Sources/scipio/scipio.docc/using-s3-storage.md

```diff
@@ -23,52 +23,55 @@ Here is a sample implementation.
 ```swift
 import ScipioS3Storage
 
-let config = S3StorageConfig(
-    authenticationMode: authorized(
-        accessKeyID: "MY_ACCESS_ID",
-        secretAccessKey: "MY_SECRET",
-        region: "ap-northeast-1",
-        endpoint: awsEndpoint,
-        shouldPublishObject: true
-    )
+let config = AuthorizedConfiguration(
+    bucket: "my-bucket",
+    region: "ap-northeast-1",
+    shouldPublishObject: true,
+    accessKeyID: "MY_ACCESS_ID",
+    secretAccessKey: "MY_SECRET"
 )
 
-let s3Storage = S3Storage(config: config)
+let s3Storage = S3Storage(config: .authorized(config))
 let runner = Runner(
     mode: .prepareDependencies,
     options: .init(
         baseBuildOptions: .init(
             buildConfiguration: .release,
             isSimulatorSupported: true
         ),
-        cacheStorage: .custom(s3Storage, [.consumer])
+        cacheMode: .storage(s3Storage, [.consumer, .producer])
     )
 )
 ```
 
 ## Configuration
 
-You can configure cache storage by `S3StorageConfig`.
+You can configure cache storage by `S3StorageConfig` and `AuthorizedConfiguration`.
 
-### Authentication Mode
+### Configuration Type
 
-You can choose authentication mode from `authorized` and `usePublicURL`.
+You can choose configuration type from `authorized` and `publicURL`.
 
 #### authorized
 
-`authorized` mode requires access key ID and secret access key.
-
-Generally, cache producer (e.g. CI job) should use this mode.
+`authorized` type is associated with `AuthorizedConfiguration` which requires the followings: 
 
-#### usePublicURL
+- bucket
+- region
+- access key ID
+- secret access key
 
-`usePubicURL` mode doesn't require any credentials.
-
-In this mode, cache system will fetch caches by public URLs.
-It's useful that developers just using caches should use this mode.
+Generally, cache producer (e.g. CI job) should use this type.
 
-### shouldPublishObject
+##### shouldPublishObject
```

Comment:
Is this level correct? 
```suggestion
#### shouldPublishObject
```

Reply:
Yes, `shouldPublishObject` is a property of `AuthorizedConfiguration`, so that should be under `#### authorized`.

https://github.com/giginet/scipio-s3-storage/blob/69a35f862ced598be9116a6ec71cb18e78d819c4/Sources/ScipioS3Storage/S3Storage.swift#L12-L23

Reply:
okay!